crowbar_framework/spec/models/service_object_spec.rb

```diff
@@ -131,5 +131,43 @@
         dns_service.validation_errors.first.should match(/cannot be assigned to both role/)
       end
     end
+
+    describe "platform" do
+      it "allows nodes of matched platform using operators" do
+        dns_service.stubs(:role_constraints).returns(
+          {
+            "dns-client" => {
+              "admin" => true ,
+              "platform" => { "ubuntu" => ">= 10" }
+            }
+          })
+        dns_service.validate_proposal_constraints(dns_proposal)
+        dns_service.validation_errors.length.should be == 0
+      end
+
+      it "allows nodes of matched platform using regular expressions" do
+        dns_service.stubs(:role_constraints).returns(
+          {
+            "dns-client" => {
+              "admin" => true ,
+              "platform" => { "ubuntu" => "/10.*/" }
+            }
+          })
+        dns_service.validate_proposal_constraints(dns_proposal)
+        dns_service.validation_errors.length.should be == 0
+      end
+
+      it "does not allow nodes of a different platform" do
+        dns_service.stubs(:role_constraints).returns(
+          {
+            "dns-client" => {
+              "admin" => true ,
+              "platform" => { "suse" => "12.0" }
+            }
+          })
+        dns_service.validate_proposal_constraints(dns_proposal)
+        dns_service.validation_errors.first.should match(/can be used only for suse 12.0/)
+      end
+    end
```

Comment:
Thanks for adding tests. Feel free to ignore my comment here, but things I would add for better coverage:

* `exclude_platform` test
* test with more than one element in `platform` (for validating that the full hash is used)
* test with a proposal where nodes with different platforms are used